File: corona-karunanagar/corona-karunanagar.cpp
#include <bits/stdc++.h>
#define ll long long

using namespace std;

int main() {
    ll t;
    cin >> t;
    while (t--) {
        ll n, d;
        string pop;
        cin >> n;
        cin >> pop;
        int connec[n];
        for (ll i = 0; i < n; i++) {
            connec[i] = 1;
        }
        cin >> d;
        ll isol[d];
        for (ll i = 0; i < d; i++) {
            cin >> isol[i];
        }

        queue<ll> q;
        if (pop[0] == '1' and pop[1] == '0') {
            q.push(0);
        }
        for (int i = 1; i < n-1; i++){
            if (pop[i] == '1' and (pop[i-1] == '0' or pop[i+1] == '0')) {
                q.push(i);
            }
        }
        if (pop[n-1] == '1' and pop[n-2] == '0') {
            q.push(n-1);
        }
        q.push(-1);

        for (ll i = 0; i < d; i++) {
            connec[isol[i] - 1] = 0;
            if (q.front() == -1) {
                break;
            }
            else {
                while (q.front() != -1) {
                    ll front = q.front();
                    if (front > 0 and pop[front - 1] == '0' and connec[front] == 1) {
                        pop[front - 1] = '1';
                        if (front > 1) {
                            if (pop[front - 2] == '0' and connec[front - 1] == 1) {
                                q.push(front - 1);
                            }
                        }
                    }
                    if (front < n-1 and pop[front + 1] == '0' and connec[front + 1] == 1) {
                        pop[front + 1] = '1';
                        if (front < n-2) {
                            if (pop[front + 2] == '0' and connec[front + 2] == 1) {
                                q.push(front + 1);
                            }
                        }
                    }
                    q.pop();
                }
                q.push(-1);
                q.pop();
            }
        }
        ll infected = 0;
        for (ll i = 0; i < n; i++) {
            if (pop[i] == '1') {
                infected++;
            }
        }
        cout << infected << "\n";
    }
    return 0;
}

File: corona-karunanagar/corona-karunanagar.java
import java.util.*;

class logic {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-->0) {
            int n = sc.nextInt();
            StringBuilder pop = new StringBuilder(sc.next());
            int connec[] = new int[n];
            Arrays.fill(connec, 1);
            int d = sc.nextInt();
            int[] isol = new int[d];
            for (int i = 0; i < d; i++) {
                isol[i] = sc.nextInt();
            }

            Queue<Integer> q = new LinkedList<>();
            if (pop.charAt(0) == '1' && pop.charAt(1) == '0') {
                q.add(0);
            }
            for (int i = 1; i < n-1; i++){
                if (pop.charAt(i) == '1' && (pop.charAt(i-1) == '0' || pop.charAt(i+1) == '0')) {
                    q.add(i);
                }
            }
            if (pop.charAt(n-1) == '1' && pop.charAt(n-2) == '0') {
                q.add(n-1);
            }
            q.add(-1);

            for (int i = 0; i < d; i++) {
                connec[isol[i] - 1] = 0;
                if (q.peek() == -1) {
                    break;
                }
                else {
                    while (q.peek() != -1) {
                        int front = q.peek();
                        if (front > 0 && pop.charAt(front - 1) == '0' && connec[front] == 1) {
                            pop.setCharAt(front - 1, '1');
                            if (front > 1) {
                                if (pop.charAt(front - 2) == '0' && connec[front - 1] == 1) {
                                    q.add(front - 1);
                                }
                            }
                        }
                        if (front < n-1 && pop.charAt(front + 1) == '0' && connec[front + 1] == 1) {
                            pop.setCharAt(front + 1, '1');
                            if (front < n-2) {
                                if (pop.charAt(front + 2) == '0' && connec[front + 2] == 1) {
                                    q.add(front + 1);
                                }
                            }
                        }
                        q.remove();
                    }
                    q.add(-1);
                    q.remove();
                }
            }
            int infected = 0;
            for (int i = 0; i < n; i++) {
                if (pop.charAt(i) == '1') {
                    infected++;
                }
            }
            System.out.println(infected);
        }
        sc.close();
    }
}
